### DISCUSSION TITLE
Seq stores floats as decimals

### DISCUSSION THREAD

**Discussion by jon-scalptrade** [datalust/seq-tickets#1944] [Help] [answered]
I recently ran into an issue where an application that logs to a Seq instance was randomly losing its connection to Seq throughout the day. For context, the application is built with C# and is using Serilog with Serilog.Sinks.Seq for connectivity. After double checking that there weren't any network issues or other obvious problems, I looked through the ingestion log and found that it was logging some exceptions, "Newtonsoft.Json.JsonReaderException: Input string '1.4698223E+36' is not a valid decimal."

It appears that when a client application tries to log a floating point outside the range of what a decimal can represent, that client application is having its connection closed. This prevents any further logs from being stored, which isn't a very desirable result. Ideally, it would be nice for Seq to just store the value as it's received, but if all floating point types are being stored as decimal then that might be a bigger issue to resolve. In the meantime, I'm wondering if there's any way to prevent this disconnect from occurring when a large float is sent over.

**Comment by nblumhardt** [comment, marked as answer]
Thanks for raising this; we're tracking it currently in https://github.com/datalust/seq-tickets/issues/1675, which we've scheduled for the next release (we need to do quite a lot of background infrastructure work to resolve it).

In the meantime, on the client-side, tracking down the data that causing the issue and using a Serilog `IDestructuringPolicy` to convert the value into a different type in the logging pipeline (i.e. to a string) would work around it. I can help with this if you can share a bit more information on what the client side looks like (logging statement, data structure...).

**Reply by jon-scalptrade** [reply]
Seems like a straightforward solution. Out of curiosity though, how would Seq handle a log message with a property that is sometimes a decimal, and sometimes a string. Would it assign them two separate Event Types, or would that property simply have an inconsistent type?

**Reply by liammclennan** [reply]
Event types are a hash of the message template, so properties are not included.
